test/ModifyTokenDistro.t.sol

```diff
@@ -0,0 +1,314 @@
+        tokenDistro.changeAddress(thirdRecipient);
+
+        // save balance values after second transfer
+        (uint256 thirdRecipientAllocatedTokensAfterTransfer,) = tokenDistro.balances(thirdRecipient);
+        (uint256 secondRecipientAllocatedTokensAfterSecondTransfer,) = tokenDistro.balances(secondRecipient);
+        // expected amount should be the sum of all three amounts
+        uint256 expectedAmount = amount1.add(amount2.add(amount3));
+        // log some stuff
+        console.log('thirdRecipientAllocatedTokensAfterTransfer: ', thirdRecipientAllocatedTokensAfterTransfer);
+        console.log('expectedAmount: ', expectedAmount);
+        // assertions
+        assertEq(thirdRecipientAllocatedTokensAfterTransfer, expectedAmount);
+        assertEq(secondRecipientAllocatedTokensAfterSecondTransfer, 0);
+    }
+
+    function testChangeAddressWithClaim(uint256 amount1, uint256 amount2) public {
+        /// @aminlatifi for some reason this does not want to work with the min bound as 1 - throws no tokens to claim error
+        amount1 = bound(amount1, 10, assignedAmount.div(2));
+        amount2 = bound(amount2, 10, assignedAmount.div(2));
+
+        address[] memory recipients = new address[](2);
+        recipients[0] = firstRecipient;
+        recipients[1] = secondRecipient;
+
+        uint256[] memory amounts = new uint256[](2);
+        amounts[0] = amount1;
+        amounts[1] = amount2;
+
+        vm.prank(distributor);
+        tokenDistro.allocateMany(recipients, amounts);
+
+        // skip ahead some time and then claim tokens
+        skip(14 days);
+        console.log('claimable for first recipient', tokenDistro.claimableNow(firstRecipient));
+        console.log('claimable for second recipient', tokenDistro.claimableNow(secondRecipient));
+
+        vm.prank(firstRecipient);
+        tokenDistro.claim();
+        vm.prank(secondRecipient);
+        tokenDistro.claim();
+
+        // save balance values
+        (, uint256 secondRecipientClaimedTokens) = tokenDistro.balances(secondRecipient);
+        (, uint256 firstRecipientClaimedTokens) = tokenDistro.balances(firstRecipient);
+        // transfer allocation to second recipient
+        vm.prank(firstRecipient);
+        tokenDistro.changeAddress(secondRecipient);
+        // check values of second recipient after transfer
+        (uint256 secondAllocatedAfterTransfer, uint256 secondClaimedAfterTransfer) =
+            tokenDistro.balances(secondRecipient);
+        (uint256 firstAllocatedAfterTransfer, uint256 firstClaimedAfterTransfer) = tokenDistro.balances(firstRecipient);
+        // assertions
+        assertEq(secondAllocatedAfterTransfer, (amount1.add(amount2)));
+        assertEq(secondClaimedAfterTransfer, (secondRecipientClaimedTokens.add(firstRecipientClaimedTokens)));
+        assertEq(firstAllocatedAfterTransfer, 0);
+        assertEq(firstClaimedAfterTransfer, 0);
+    }
+
+    // function testCancelAllocation() public {
```

Comment:
as a good practice lets remove commented code. we can always access it from github history when needed.